compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -4078,6 +4078,8 @@ private BLangOnFailClause createTrxInternalOnFail(Location pos, BLangSimpleVarRe
 
     @Override
     public void visit(BLangTransaction transactionNode) {
+        boolean previousInTransactionContext = this.inTransactionContext;
+        this.inTransactionContext = true;
```

Comment:
without checking whether we are in a trx scope here, shall we check it in CodeAnalyzer's ```visit(BLangInvocation.BLangActionInvocation actionInvocation)``` ? We have this check there already

Reply:
Fixed